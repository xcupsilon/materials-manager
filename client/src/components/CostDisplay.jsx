import React from 'react';

const CostDisplay = ({ matVolume, matCost }) => (
  <div className="flex justify-between w-[17rem] text-text mt-[1.5rem]">
    <div>
      Total Cost:
    </div>
    <div>
      ${(matCost * matVolume).toFixed(2)}
    </div>
  </div>
)

export default CostDisplay