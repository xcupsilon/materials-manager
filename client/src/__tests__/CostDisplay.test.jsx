/**
 * @jest-environment jsdom
 */

import { render, cleanup, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import CostDisplay from '../components/CostDisplay'

afterEach(cleanup)

it('renders', () => {
  const { asFragment } = render(<CostDisplay matCost={100} matVolume={100} />)
  expect(asFragment()).toMatchSnapshot()
})

it('correctly calculates and inserts into div', () => {
  render(<CostDisplay matCost={100} matVolume={100} />)

  expect(screen.getByTestId('totalcostdiv')).toHaveTextContent(10000.00)
})