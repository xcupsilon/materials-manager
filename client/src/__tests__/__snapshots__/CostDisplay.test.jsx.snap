// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between w-[17rem] text-text mt-[1.5rem]"
  >
    <div>
      Total Cost:
    </div>
    <div
      data-testid="totalcostdiv"
    >
      $10000.00
    </div>
  </div>
</DocumentFragment>
`;
